torch/nn/init.py

```diff
@@ -384,6 +384,9 @@ def kaiming_uniform_(tensor, a=0, mode='fan_in', nonlinearity='leaky_relu'):
         >>> w = torch.empty(3, 5)
         >>> nn.init.kaiming_uniform_(w, mode='fan_in', nonlinearity='relu')
     """
+    if 0 not in tensor.shape:
```

Comment:
Logic seems reversed here and below. Also maybe checking `tensor.numel() == 0` instead is better?